src/main/java/com/google/ehub/data/CommentDataManager.java

```diff
@@ -61,17 +64,26 @@ public void addItemComment(long itemId, String comment, long timestampMillis, St
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery comments = datastore.prepare(itemCommentQuery);
     for (Entity entity : comments.asIterable()) {
-      String email = (String) entity.getProperty(EMAIL_PROPERTY_KEY);
-      UserProfile userProfile = profileData.getUserProfile(email);
-      if(userProfile == null){
+      String storedEmail = (String) entity.getProperty(EMAIL_PROPERTY_KEY);
+      UserProfile userProfile = profileData.getUserProfile(storedEmail);
+      if (userProfile == null) {
         continue;
       }
       String comment = (String) entity.getProperty(COMMENT_PROPERTY_KEY);
       long timestampMillis = (Long) entity.getProperty(TIMESTAMP_PROPERTY_KEY);
-      
-      String username = userProfile.getUsername(); 
-      results.add(new CommentData(itemId, comment, timestampMillis, username));
+
+      String username = userProfile.getUsername();
+      long commentId = entity.getKey().getId();
+      String currentEmail = userService.getCurrentUser().getEmail();
+      boolean belongsToUser = storedEmail.equals(currentEmail);
+      results.add(
+          new CommentData(itemId, comment, timestampMillis, username, commentId, belongsToUser));
     }
-   return results;
-}
+    return results;
+  }
+
+  public void deleteComment(long commentId, String COMMENT_KIND_KEY) {
```

Comment:
`COMMENT_KIND_KEY` -> `commentKindKey`
Or its better to just remove this parameter and directly use the constant COMMENT_KIND_KEY in the method.